code/game/objects/structures/lavaland/necropolis_tendril.dm

```diff
@@ -42,6 +44,26 @@ GLOBAL_LIST_EMPTY(tendrils)
 	new /obj/structure/closet/crate/necropolis/tendril(loc)
 	return ..()
 
+/obj/structure/spawner/lavaland/proc/rally_towards(mob/living/target)
+	var/datum/component/spawner/spawn_component = GetComponent(/datum/component/spawner)
```

Comment:
According to how components work, we should be minimizing use of `GetComponent`. Could you instead put this on the component and a new signal, maybe something like `COMSIG_SPAWNER_SET_TARGET`, and send that signal on where you currently have `rally_towards`?

Reply:
Will attempt this, for now it's held back by the fact that the targeting code for hostile mobs has a blocking call on `tgui_input_text` for emoting. There may also be another issue where megafauna, who are all hostile subtypes despite not being meant for this code, use `SLEEP_CHECK_DEATH`, another thing not permitted in a signal handler.

Reply:
Yea, you'll have to use some INVOKE_ASYNC calls

Reply:
Done and tested, seems to work well, PTAL?